x/ecocredit/client/tx.go

```diff
@@ -504,3 +509,224 @@ Parameters:
+				}
+
+				// parse and normalize new ask price as sdk.Coin
+				askPrice, err := sdk.ParseCoinNormalized(update.NewAskPrice)
+				if err != nil {
+					return err
+				}
+
+				// set update with new ask price as sdk.Coin
+				updates[i] = &ecocredit.MsgUpdateSellOrders_Update{
+					SellOrderId:       sellOrderId,
+					NewAskPrice:       &askPrice,
+					NewQuantity:       update.NewQuantity,
+					DisableAutoRetire: update.DisableAutoRetire,
+				}
+			}
+
+			// create update sell orders message
+			msg := ecocredit.MsgUpdateSellOrders{
+				Owner:   owner.String(),
+				Updates: updates,
+			}
+
+			// generate and broadcast transaction
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	})
+}
+
+// TxBuyCmd returns a transaction command that creates sell orders.
+func TxBuyCmd() *cobra.Command {
+	return txflags(&cobra.Command{
+		Use:   "buy [orders]",
+		Short: "Creates new buy orders with transaction author (--from) as buyer",
+		Long: fmt.Sprintf(
+			`Creates new buy orders with transaction author (--from) as buyer.
+
+Parameters:
+  orders:  YAML encoded order list. Note: numerical values must be written in strings.
+           eg: '[{sell_order_id: "1", quantity: "5", bid_price: "100regen", disable_auto_retire: false}]'`,
+		),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := sdkclient.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			// get the order buyer from the --from flag
+			buyer := clientCtx.GetFromAddress()
+
+			// declare orders array with bid price as string
+			var strOrders []struct {
+				SellOrderId       string `json:"sell_order_id"`
+				Quantity          string `json:"quantity"`
+				BidPrice          string `json:"bid_price"`
+				DisableAutoRetire bool   `json:"disable_auto_retire"`
+			}
+
+			// unmarshal YAML encoded orders with new bid price as string
+			if err := yaml.Unmarshal([]byte(args[0]), &strOrders); err != nil {
+				return err
+			}
+
+			// declare orders array with new bid price as sdk.Coin
+			orders := make([]*ecocredit.MsgBuy_Order, len(strOrders))
+
+			// loop through orders with new bid price as string
+			for i, order := range strOrders {
+
+				// parse sell order id
+				sellOrderId, err := strconv.ParseUint(order.SellOrderId, 10, 64)
+				if err != nil {
+					return ecocredit.ErrInvalidSellOrder.Wrap(err.Error())
+				}
+
+				// set sell order id as buy order selection
+				selection := &ecocredit.MsgBuy_Order_Selection{
+					Sum: &ecocredit.MsgBuy_Order_Selection_SellOrderId{SellOrderId: sellOrderId},
+				}
+
+				// parse and normalize new bid price as sdk.Coin
+				bidPrice, err := sdk.ParseCoinNormalized(order.BidPrice)
+				if err != nil {
+					return err
+				}
+
+				// set order with new bid price as sdk.Coin
+				orders[i] = &ecocredit.MsgBuy_Order{
+					Selection:         selection,
+					BidPrice:          &bidPrice,
+					Quantity:          order.Quantity,
```

Comment:
verify here

Reply:
We already do in the message implementation. I don't think we need to duplicate the verification here. See other tx commands for example and test cases within this pull request.